doc/pre-transaction-actions.rst

```diff
@@ -0,0 +1,97 @@
+..
+  Copyright (C) 2019 Red Hat, Inc.
+
+  This copyrighted material is made available to anyone wishing to use,
+  modify, copy, or redistribute it subject to the terms and conditions of
+  the GNU General Public License v.2, or (at your option) any later version.
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY expressed or implied, including the implied warranties of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+  Public License for more details.  You should have received a copy of the
+  GNU General Public License along with this program; if not, write to the
+  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+  02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+  source code or documentation are not subject to the GNU General Public
+  License and may only be used or replicated with the express permission of
+  Red Hat, Inc.
+
+===================================
+DNF pre-transaction-actions Plugin
+===================================
+
+-----------
+Description
+-----------
+
+The plugin allows to define actions to be executed upon starting an RPM transaction. Each action
+may define a (glob-like) filtering rule on the package NEVRA or package files, as well as whether
+the package was installed or removed. Actions are defined in action files.
+
+-------------
+Configuration
+-------------
+
+The plugin configuration is in ``/etc/dnf/plugins/pre-transaction-actions.conf``. All configuration
+options are in the ``[main]`` section.
+
+``enabled``
+    Whether the plugin is enabled. Default value is ``True``.
+
+``actiondir``
+    Path to the directory with action files. Action files must have the ".action" extension.
+    Default value is "/etc/dnf/plugins/pre-transaction-actions.d/".
+
+------------------
+Action file format
+------------------
+
+Empty lines and lines that start with a '#' character are ignored.
+Each non-comment line defines an action and consists of three items separated by colons:
+``package_filter:transaction_state:command``.
+
+``package_filter``
+   A (glob-like) filtering rule applied on the package NEVRA (also in the shortened forms) or
+   package files.
+
+``transaction_state``
+   Filters packages according to their state in the transaction.
+
+   * ``install`` - packages that will be instaled on the system (install)
```

Comment:
As I wrote in the review of the source code. Replace "install", "update" and "remove" with "in" and "out".

Based on the description of the "post-transaction-actions" plugin something like this:
   * ``in`` - packages that will appear on the system (downgrade, install, obsolete, reinstall, upgrade)
   * ``out`` - packages that will disappear from the system (downgraded, obsoleted, remove, upgraded)